### PR TITLE
allow info, update, detach and delete commands on multiple ips

### LINKED ISSUE #129: [Feature request] Ability to delete a list of IP addresses
I tried to delete a bunch of private IP addresses by running 

``` sh
gandi ip delete --force $(gandi ip list --detached | awk '/ip/ {print $3}')
```

and encounter the following error:

> Error: Got unexpected extra arguments (192.168.1.4 192.168.1.5 )

It would be great to be able to delete IP addresses by batch, don't you think ?

To bypass this limitation, I temporary use a bash `for` loop:

``` sh
for i in $(gandi ip list --detached | awk '/ip/ {print $3}') ; do gandi ip delete --force $i ; done
```

Here is my current configuration

```
$ uname -a
Linux XXXXX 3.14-2-amd64 #1 SMP Debian 3.14.15-2 (2014-08-09) x86_64 GNU/Linux

$ gandi --version
Gandi CLI 0.12

Copyright: © 2014-2015 Gandi S.A.S.
License: GPL-3
```

### PR DESCRIPTION
**Pull request by soulshake**
Fixes #129

### CODE REVIEW

File: gandi/cli/commands/ip.py [hunk 1]
```diff
@@ -77,32 +77,38 @@ def list(gandi, datacenter, type, id, attached, detached, version, reverse,
...
+        output_ip(gandi, ip, datacenters, vms, {iface['id']: iface},
+                  output_keys)
 
-    return ip
+    return ret
 
 
 @cli.command()
-@click.argument('ip')
+@click.argument('ip', nargs=-1)
```
Comment (sayoun) [inline gandi/cli/commands/ip.py]: You should not use multiple args here in `update` command as we can only update one reverse per ip, I think that it makes no sense to update the same reverse for different ips (but I may be missing a use case here).

File: gandi/cli/commands/ip.py [hunk 2]
```diff
@@ -123,32 +129,44 @@ def update(gandi, ip, reverse, background):
...
-                return
+    resources = sorted(tuple(set(resources)))
+    ret = []
+
+    for num, item in enumerate(resources):
+        try:
+            ip_ = gandi.ip.info(item)
+            vm_ = gandi.iaas.info(vm)
+        except UsageError:
+            gandi.error("Can't find this ip %s" % ip)
```
Comment (sayoun) [inline gandi/cli/commands/ip.py]: ip parameter does not exists anymore, should be `item` for this error

File: gandi/cli/commands/ip.py [hunk 3]
```diff
@@ -123,32 +129,44 @@ def update(gandi, ip, reverse, background):
...
+            break
+
+        iface = gandi.iface.info(ip_['iface_id'])
+        if iface.get('vm_id'):
+            if vm_ and iface['vm_id'] == vm_.get('id'):
+                gandi.echo('IP %s is already attached to vm %s.' %
+                           (ip_['ip'], iface['vm_id']))
+                continue
+
+            if not force:
```
Comment (sayoun) [inline gandi/cli/commands/ip.py]: You should check for `force` args and ask for prompt outside this loop on resources, so you can only ask one time the user and not on every loop iteration. You can see how it's been done in `vm delete` command for inspiration.

File: gandi/cli/commands/ip.py [hunk 4]
```diff
@@ -123,32 +129,44 @@ def update(gandi, ip, reverse, background):
...
+                proceed = click.confirm('Are you sure you want to detach'
+                                        ' %s from vm %s?' %
+                                        (ip_['ip'], iface['vm_id']))
+                if not proceed:
+                    continue
+
+        gandi.echo('Attaching IP %s to VM %s...' %
+                    (ip_['ip'], iface['vm_id']))
+
+        ret.append(gandi.ip.attach(item, vm, background, force))
```
Comment (sayoun) [inline gandi/cli/commands/ip.py]: Same here for the call to `gandi.ip.attach`, it will be hard to follow as it will made a progress bar per ip attach operations, You should update the module `IP.attach` method so it can handle multiple operations in input, as done in the vm module, so you can have one progress bar for all operations.

File: gandi/cli/commands/ip.py [hunk 5]
```diff
@@ -199,52 +217,65 @@ def create(gandi, datacenter, bandwidth, ip_version, vlan, ip, attach,
...
-    if not force:
-        proceed = click.confirm('Are you sure to detach ip %s?' % resource)
-        if not proceed:
-            return
+    resources = sorted(tuple(set(resources)))
+    ret = []
 
-    return gandi.ip.detach(resource, background, force)
+    for num, item in enumerate(resources):
+        if not force:
```
Comment (sayoun) [inline gandi/cli/commands/ip.py]: same remarks as in attach, you should check for force outside the loop, and group the operations into one `IP.detach` call.

File: gandi/cli/commands/ip.py [hunk 6]
```diff
@@ -199,52 +217,65 @@ def create(gandi, datacenter, bandwidth, ip_version, vlan, ip, attach,
...
+            ip_ = gandi.ip.info(item)
+        except UsageError:
+            gandi.error("Can't find ip %s" % item)
+
+        iface = gandi.iface.info(ip_['iface_id'])
+        ips = ', '.join([ip['ip'] for ip in iface['ips']])
+        if len(iface['ips']) > 1:
+            gandi.echo('None of the ips (%s) is attached to a vm; will delete them all' %
+                     ips)
+        if not force:
```
Comment (sayoun) [inline gandi/cli/commands/ip.py]: check for `force` outisde the loop, group the operations :)

**Comment by soulshake** [comment]
Sorry this stagnated. I haven't had time to come back to it, so I'm closing this PR for now. Hopefully I can revisit this in the future in a new PR.
